File: Client/src/constants/jewelry.ts
import { ICartType } from "@/types/cart.interface";
import { IJewelry, IJewelryStatus } from "@/types/jewelry.interface";

export const jewelryData: IJewelry[] = [
  {
    productType: ICartType.Jewelry,
    jewelryId: "J002",
    jewelryCategory: "Earrings",
    jewelryName: "18K DIAMOND EARRINGS",
    image: "https://jemmia.vn/wp-content/uploads/2024/04/1-copy-8.jpg",
    image2: "https://jemmia.vn/wp-content/uploads/2024/04/1-copy-8.jpg",
    mainStoneSize: "Round 5ly",
    sideStoneType: "Diamond",
    sideStoneQuantity: 36,
    sideStoneWeight: 0.18,
    goldType: "White gold",
    goldKarat: "18K",
    goldWeight: "0.86",
    price: 20210400,
    status: IJewelryStatus.Available,
  },
];

File: Server/DiamonShop.Core/Domain/Content/Jewelry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DiamonShop.Core.Domain.Content
{
    public class Jewelry
    {
        [Key, ForeignKey(nameof(Product))]
        public Guid JewelryId { get; set; }

        public string SkuID { get; set; }

        [Required]
        public string Name { get; set; }
        public string ProductType { get; set; }

        [Required]

        public string MainStoneSize { get; set; }
        [Required]

        public string sideStoneType { get; set; }
        [Required]

        public int sideStoneQuantity { get; set; }
        [Required]

        public double StoneWeight { get; set; }
        [Required]

        public string GoldType { get; set; }
        [Required]

        public string GoldKarat { get; set; }
        [Required]
        public double GoldWeight { get; set; }
        [Required]
        public double Price { get; set; }

        //navigate 
        public Product Product { get; set; }
    }
}
